lib/classroom/models/searching.rb

```diff
@@ -15,24 +14,40 @@ def pipeline
 
   class StudentFilter < BaseFilter
     def query
-      {'$text': {'$search': @query_param}}
+      {'$text': {'$search': query_param}}
+    end
+  end
+
+  class NumericFilter < BaseFilter
+    def query_param=(query_param)
+      @query_param = query_param.to_i
     end
   end
 
   def self.default_filter
     StudentFilter
   end
 
-  def self.filter_for(criteria, collection, query)
+  def self.filter_for(criteria, collection, query, query_operand)
```

Comment:
If we are going to change this signature, isn't it better to change it to receive a hash? Or even the query object?